<?xml version="1.0"?>
<launch>

    <arg name="testName" default="dji_test"/>

    <node pkg="octomap_server" type="octomap_server_node" name="octomap_server"
	    args="$(find lidar_eskf)/map/$(arg testName).bt">
	    <param name="frame_id" value="world"/>
    </node>

    <node pkg="lidar_eskf" type="lidar_eskf_node" name="lidar_eskf" output="screen">
	
        <remap from="/lidar_eskf/cloud"        to="/sync_cloud"/>
	    <remap from="/lidar_eskf/scan"         to="/sync_scan"/>
        <remap from="/imu"                     to="/microstrain/imu"/>
	    <remap from="/meas"                    to="/lidar_eskf/posterior"/>
	    
        <param name="map_file_name"            value="$(find lidar_eskf)/map/$(arg testName).bt"/>
        <param name="octree_resolution"        value="0.1"/>
        <param name="max_obstacle_dist"        value="2.0"/>
        <param name="ray_sigma"                value="1.0"/>
        <param name="cloud_resolution"         value="0.1"/>
        <param name="cloud_range"              value="30.0"/>
        <param name="laser_type"               value="pointcloud"/>
        <param name="pcd_file"                 value="$(find lidar_eskf)/dat/rec_$(arg testName).pcd"/>
        <param name="set_size"                 value="500"/>

        <param name="imu_frequency"             value="50"/>
        <param name="sigma_acceleration"        value="0.8"/>
        <param name="sigma_gyroscop"            value="0.1"/>
        <param name="sigma_acceleration_bias"   value="0.02"/>
        <param name="sigma_gyroscope_bias"      value="0.01"/>
        <param name="gravity"                   value="0.0"/>
        
        <param name="init_bias_acc_x"           value="0.0"/> #-0.05
        <param name="init_bias_acc_y"           value="0.0"/> #0.025
        <param name="init_bias_acc_z"           value="0.0"/> #-0.032
        
        <param name="acc_queue_size"            value="3"/>
    </node>
</launch>
